src/EditorFeatures/CSharp/QuickInfo/SyntacticQuickInfoProvider.cs

```diff
@@ -76,8 +85,66 @@ internal class SyntacticQuickInfoProvider : AbstractQuickInfoProvider
                 return null;
             }
 
-            var span = new SnapshotSpan(textSnapshot, Span.FromBounds(parent.SpanStart, openBrace.Span.End));
+            var span = new SnapshotSpan(textSnapshot, Span.FromBounds(spanStart, spanEnd));
             return this.CreateElisionBufferDeferredContent(span);
         }
+
+        private static void MarkInterestedSpanNearbyScopeBlock(SyntaxNode block, SyntaxToken openBrace, ref int spanStart, ref int spanEnd)
+        {
+            SyntaxTrivia nearbyTrivia;
+
+            if (openBrace.HasLeadingTrivia && FindFurthestNearbyComment(openBrace.LeadingTrivia.Reverse(), out nearbyTrivia))
+            {
+                spanStart = nearbyTrivia.SpanStart;
+                return;
+            }
+
+            var nextToken = block.FindToken(openBrace.FullSpan.End);
+            if (nextToken.HasLeadingTrivia && FindFurthestNearbyComment(nextToken.LeadingTrivia, out nearbyTrivia))
+            {
+                spanEnd = nearbyTrivia.Span.End;
+                return;
+            }
+        }
+
+        private static bool FindFurthestNearbyComment(IEnumerable<SyntaxTrivia> triviaSearchList, out SyntaxTrivia nearbyTrivia)
+        {
+            var searchList = triviaSearchList.SkipWhile(IsIndentation).ToArray();
```

Comment:
To avoid allocation and the boxing passing in the first parameter, I have another implementation of this method, but it looks rather ugly as it uses state machine, goto..
```C#
        const int MarkIfMultiLineComment = 1;
        const int MarkIfSingleLineComment = 2;
        const int CrawlFurtherIfPreviouslyMarkedSingleLineComment = 3;

        private static bool FindFurthestNearbyComment<T>(ref T triviaSearchList, out SyntaxTrivia nearbyTrivia)
            where T : IEnumerable<SyntaxTrivia>
        {
            nearbyTrivia = default(SyntaxTrivia);

            var skipped = 0;
            var state = MarkIfMultiLineComment;

            foreach (var trivia in triviaSearchList)
            {
                switch (state)
                {
                    case MarkIfMultiLineComment:

                        if (IsIndentation(trivia, skipped))
                        {
                            skipped++;
                            continue;
                        }

                        if (trivia.IsKind(SyntaxKind.MultiLineCommentTrivia))
                        {
                            nearbyTrivia = trivia;
                            return true;
                        }

                        goto case MarkIfSingleLineComment;

                    case MarkIfSingleLineComment:

                        if (trivia.IsKind(SyntaxKind.SingleLineCommentTrivia))
                        {
                            nearbyTrivia = trivia;

                            skipped = 0;
                            state = CrawlFurtherIfPreviouslyMarkedSingleLineComment;
                            continue;
                        }

                        break;

                    case CrawlFurtherIfPreviouslyMarkedSingleLineComment:

                        if (IsIndentation(trivia, skipped))
                        {
                            skipped++;
                            continue;
                        }

                        goto case MarkIfSingleLineComment;

                    default:
                        break;
                }
            }

            return nearbyTrivia.IsKind(SyntaxKind.SingleLineCommentTrivia);
        }
```